lib/payment_processor/braintree/transaction.rb

```diff
@@ -18,40 +18,49 @@ class Transaction < Populator
       # * +:currency+ - Billing currency (required)
       # * +:user+     - Hash of information describing the customer. Must include email, and name (required)
       # * +:customer+ - Instance of existing Braintree customer. Must respond to +customer_id+ (optional)
-      attr_reader :action, :result, :store_in_vault
+      attr_reader :action, :result
 
-      def self.make_transaction(nonce:, amount:, currency:, user:, page_id:, store_in_vault: false, device_data: {})
-        builder = new(nonce, amount, currency, user, page_id, store_in_vault, device_data)
+      def self.make_transaction(nonce:, amount:, currency:, user:, page:, store_in_vault: false, device_data: {})
+        builder = new(nonce, amount, currency, user, page, store_in_vault, device_data)
         builder.transact
         builder
       end
 
       # Long parameter list is doing my head in - let's replace with a parameter object
       #
-      def initialize(nonce, amount, currency, user, page_id, store_in_vault = false, device_data = {})
+      def initialize(nonce, amount, currency, user, page, store_in_vault = false, device_data = {})
         @amount = amount
         @nonce = nonce
         @user = user
         @currency = currency
-        @page_id = page_id
+        @page = page
         @store_in_vault = store_in_vault
         @device_data = device_data
       end
 
       def transact
         @result = ::Braintree::Transaction.sale(options)
+
         if @result.success?
-          @action = ManageBraintreeDonation.create(params: @user.merge(page_id: @page_id), braintree_result: @result, is_subscription: false, store_in_vault: @store_in_vault)
-          Payment::Braintree.write_transaction(@result, @page_id, @action.member_id, existing_customer, store_in_vault: @store_in_vault)
+          @action = ManageBraintreeDonation.create(params: @user.merge(page_id: @page.id), braintree_result: @result, is_subscription: false, store_in_vault: @store_in_vault)
+          Payment::Braintree.write_transaction(@result, @page, @action.member_id, existing_customer, store_in_vault: @store_in_vault)
         else
-          Payment::Braintree.write_transaction(@result, @page_id, nil, existing_customer, store_in_vault: @store_in_vault)
+          Payment::Braintree.write_transaction(@result, @page, nil, existing_customer, store_in_vault: @store_in_vault)
         end
       end
 
       def transaction_id
         @result.try(:transaction).try(:id)
       end
 
+      def store_in_vault?
+        @store_in_vault || @page.pledger?
```

Comment:
no - it must go. we've no reason to store the payment method for delayed transaction settlement. Thanks for the catch.